datadog_checks_dev/datadog_checks/dev/tooling/commands/docs/deploy.py

```diff
@@ -0,0 +1,87 @@
+from ...constants import get_root
+from ...git import get_git_email, get_git_user, get_latest_commit_hash
+from ..console import CONTEXT_SETTINGS, abort, echo_info, echo_success, echo_waiting, echo_warning, run_or_abort
+
+PRODUCTION_BRANCH = 'gh-pages'
+
+
+@click.command(context_settings=CONTEXT_SETTINGS, short_help='Deploy built documentation')
+@click.argument('branch', required=False)
+@click.option('--yes', '-y', is_flag=True)
+@click.pass_context
+def deploy(ctx, branch, yes):
+    """Deploy built documentation."""
+    # We allow specifying a branch in case you want to quickly showcase changes to others
+    if not branch:
+        branch = PRODUCTION_BRANCH
+
+    github_token = ctx.obj['github'].get('token')
+    if not github_token:
+        abort('No `github.token` has been set')
+
+    site_dir = path_join(get_root(), 'site')
+    if not dir_exists(site_dir):
+        abort('Site directory does not exist, build docs by running `ddev docs build`')
+
+    echo_waiting('Reading current Git configuration...')
+    git_user = get_git_user()
+    git_email = get_git_email()
+    latest_commit_hash = get_latest_commit_hash()
+    repo_name = ctx.obj['repo_name']
+
+    if 'GITHUB_ACTIONS' in os.environ:
+        remote = f'https://{os.getenv("GITHUB_ACTOR")}:{github_token}@github.com/DataDog/{repo_name}.git'
+    else:
+        remote = f'https://{github_token}@github.com/DataDog/{repo_name}.git'
+
+    echo_waiting('Copying site to a temporary directory...')
+    with temp_dir() as d:
+        temp_repo_dir = copy_path(site_dir, d)
+
+        # https://help.github.com/en/github/working-with-github-pages/about-github-pages#static-site-generators
+        # https://github.com/mkdocs/mkdocs/pull/2060
+        echo_waiting('Writing .nojekyll at the root...')
+        create_file(path_join(temp_repo_dir, '.nojekyll'))
+
+        with chdir(temp_repo_dir):
+            echo_waiting('Configuring the temporary Git repository...')
+            run_or_abort('git init', capture=True)
+            run_or_abort(f'git config user.name "{git_user}"', capture=True)
+            run_or_abort(f'git config user.email "{git_email}"', capture=True)
+            run_or_abort(f'git remote add upstream {remote}', capture=True)
+
+            echo_waiting('Discovering remote...')
+            run_or_abort('git fetch --depth 1 upstream', capture=True)
+
+            upstream_check = run_or_abort(f'git ls-remote --heads {remote} {branch}', capture=True)
+            if upstream_check.stdout.strip():
+                run_or_abort(f'git reset upstream/{branch}', capture=True)
+
+            echo_waiting(f'Committing site contents to branch {branch}...')
+            run_or_abort('git add --all', capture=True)
+
+            # If no changes, degrade gracefully
+            result = run_or_abort(
+                f'git commit -m "build docs at {latest_commit_hash}"', capture=True, ignore_exit_code=True
+            )
```

Comment:
Just a heads up FYI: I have my git setup to commit with my DD account and GPG keys inside `~/dd`, and my personal account (and keys) elsewhere, and this command makes git use my personal account.

I'm not sure why. Obviously we're not inside `~/dd` at this point (but in a random tmp dir), but the `git config` setup done above should override my `~/.gitconfig-*`, right? I also tried passing `git commit --author "{git_user} <{git_email}>"` but it doesn't take it into account either.

Not a big issue but if you have any clues I'd be interested. :)

Reply:
Hmm that's interesting. I'm not sure